src/app/daemon.cc

```diff
@@ -81,6 +81,7 @@ bool DaemonSingleton::Init() {
     m_Instance->Initialize();
   } catch (const std::exception& ex) {
     LOG(error) << "DaemonSingleton: exception during initialization: " << ex.what();
+    Stop();
```

Comment:
>Not sure it's the best fix

Yes, I agree. We shouldn't need to run this call if we've technically never `Start`ed 😏 

For the time being, will you consider applying at least one of these two options (or both)?:

1. Adding a quick check for `host` validity during instance *configuration* and bail there if `host` is invalid (can be ensured with members from `boost::asio::ip::address`)
2. Adding a TODO above this line (Instance and RAII refactoring is unfinished)

Any other ideas for a quick fix?